src/buffer.h

```diff
@@ -156,67 +150,15 @@ class Buffer {
   /// Succeeds only if both buffer contents are equal
   Result IsEqual(Buffer* buffer) const;
 
- protected:
-  Buffer();
-
-  std::vector<uint8_t> values_;
-  std::unique_ptr<Format> format_;
-
  private:
   BufferType buffer_type_ = BufferType::kUnknown;
   std::string name_;
   uint32_t element_count_ = 0;
   uint32_t width_ = 0;
   uint32_t height_ = 0;
   uint8_t location_ = 0;
-};
-
-/// A buffer class where the data is described by a |DatumType| object.
-class DataBuffer : public Buffer {
- public:
-  DataBuffer();
-  explicit DataBuffer(BufferType type);
-  ~DataBuffer() override;
-
-  // Buffer
-  bool IsDataBuffer() const override { return true; }
-  Result SetData(const std::vector<Value>& data) override;
-
-  /// Sets the DatumType of the buffer to |type|.
-  void SetDatumType(const DatumType& type) {
-    datum_type_ = type;
-    format_ = datum_type_.AsFormat();
-  }
-  /// Returns the DatumType describing the buffer data.
-  const DatumType GetDatumType() const { return datum_type_; }
-
- private:
-  Result CopyData(const std::vector<Value>& data);
-
-  DatumType datum_type_;
-};
-
-/// A buffer class where the data is described by a |format| object.
-class FormatBuffer : public Buffer {
- public:
-  FormatBuffer();
-  explicit FormatBuffer(BufferType type);
-  ~FormatBuffer() override;
-
-  // Buffer
-  bool IsFormatBuffer() const override { return true; }
-
-  Result SetData(const std::vector<Value>& data) override;
-
-  uint32_t GetTexelStride() { return format_->SizeInBytes(); }
-
-  // When copying the image to the host buffer, we specify a row length of 0
-  // which results in tight packing of rows.  So the row stride is the product
-  // of the texel stride and the number of texels in a row.
-  uint32_t GetRowStride() { return GetTexelStride() * GetWidth(); }
-
- private:
-  Result CopyData(const std::vector<Value>& data);
+  std::vector<uint8_t> values_;
```

Comment:
values_ is a deceptive name because we have a Value class which is a different thing.
Should this be bytes_ ?